app/services/user/bulk_destroy.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class User::BulkDestroy
+  include Service::Base
+
+  params do
+    attribute :user_ids, :array
+
+    validates :user_ids, length: { maximum: 100 }
+  end
+
+  model :users
+  policy :can_delete_users
+  step :delete
+
+  private
+
+  def fetch_users(params:)
+    ids = params.user_ids.to_a
+    # this order cluase ensures we retrieve the users in the same order as the
+    # IDs in the param. we do this to ensure the users are deleted in the same
+    # order as they're selected in the UI
+    User.where(id: ids).order(DB.sql_fragment("array_position(ARRAY[?], users.id)", ids))
+  end
+
+  def can_delete_users(guardian:, users:)
+    users.all? { |u| guardian.can_delete_user?(u) }
+  end
+
+  def delete(users:, guardian:)
+    users.each.with_index do |u, index|
+      position = index + 1
+      success =
+        UserDestroyer.new(guardian.user).destroy(
+          u,
+          delete_posts: true,
+          prepare_for_destroy: true,
+          context: I18n.t("staff_action_logs.bulk_user_delete", users: users.map(&:id).inspect),
```

Comment:
I think it's unnecessary to add all the user IDs as context, there is a lot of repetition and also you are doing map for every single user in the array

Reply:
oh my bad, removed